src/server/session.ts

```diff
@@ -1690,284 +1707,188 @@ namespace ts.server {
         exit() {
         }
 
-        private notRequired(): HandlerResponse {
-            return { responseRequired: false };
-        }
-
         private requiredResponse(response: {}): HandlerResponse {
             return { response, responseRequired: true };
         }
 
+        private respondTrue(cb: () => void): HandlerResponse {
+            cb();
+            return this.requiredResponse(/*response*/ true);
+        }
+
+        private noResponse(cb: () => void): HandlerResponse {
+            cb();
+            return { responseRequired: false };
+        }
+
         private handlers = createMapFromTemplate<(request: protocol.Request) => HandlerResponse>({
-            [CommandNames.OpenExternalProject]: (request: protocol.OpenExternalProjectRequest) => {
+            [CommandNames.OpenExternalProject]: (request: protocol.OpenExternalProjectRequest) => this.respondTrue(() => {

```

Comment:
seems a bit weird to add a new dispatch just to make the linter happy..

Reply:
+1 It also means we are creating unnecessary function expression per handler..

Reply:
do we even need this lint rule...